### ISSUE TITLE
JupyterLab not running interactive plots with %matplotlib notebook

### ISSUE THREAD

**Issue by seredes** [jupyterlab/jupyterlab#10198]
**Python version**: Python 3.9.2
**JupyterLab version**: 3.0.14
**Notebook version**: 
jupyter core     : 4.7.1
jupyter-notebook : 6.3.0
qtconsole        : not installed
ipython          : 7.23.0
ipykernel        : 5.5.3
jupyter client   : 6.1.12
jupyter lab      : 3.0.14
nbconvert        : 6.0.7
ipywidgets       : 7.6.3
nbformat         : 5.1.3
traitlets        : 5.0.5

**browser versions affected**:
**Safari** - Version 14.0.3 (14610.4.3.1.7)
**Chrome** - Version 90.0.4430.93 (Official Build) (x86_64)
**Firefox** - 86.0.1 (64-bit)

**operating system and version**: macOS Mojave 10.14.6

**Setup**

I created a clean environment first:

```
conda create -n ipywidgets-test --override-channels --strict-channel-priority -c conda-forge -c anaconda jupyterlab
conda activate ipywidgets-test
pip install --upgrade ipywidgets
```
Then I installed the following packages:
```
pandas
matplotlib
seaborn
scikit-learn
```
**Problem**

I run my code, then I use the command:
```
%matplotlib notebook
```
After that:
```
fig = plt.figure(figsize=(8,6))
```
That's when I get the following error message:
```
Javascript Error: IPython is not defined
```
The error does not appear on a classic Jupyter notebook.
Following [this answer on SO](https://stackoverflow.com/a/55848505/5535152), I have tried to implement the following:

```
conda install -y nodejs
pip install ipympl
pip install --upgrade jupyterlab
jupyter labextension install @jupyter-widgets/jupyterlab-manager
jupyter labextension install jupyter-matplotlib
jupyter nbextension enable --py widgetsnbextension
```
I still get the same error.

In a nutshell, this is the error I am getting:
<img width="605" alt="image" src="https://user-images.githubusercontent.com/20558980/117223249-4448d900-addb-11eb-878a-8f7b8b48230f.png">

**Comment by agoose77** [comment]
@seredes I believe the `notebook` backend is designed for Jupyter Notebook, which was a bit of a wild-west when it comes to running arbitrary Javascript. The new JupyterLab extension (`ipympl`) adds proper support for Matplotlib, but you need to request its backend with `%matplotlib widget`. Let me know if that fixes your issue.

**Comment by seredes** [comment]
Thanks @agoose77, I sorted it out using `ipympl` and `widget`. But I also had to install `nodejs` as highlighted in my post.

**Comment by agoose77** [comment]
I'm glad you've sorted it! You shouldn't need nodejs or to install any
extensions via jupyterlab in the new jupyterlab version. Everything is
installed by the pip packages!
